sympy/stats/stochastic_process_types.py

```diff
@@ -94,7 +95,7 @@ def _sym_sympify(arg):
         return _sympify(arg)
 
 def _matrix_checks(matrix):
-    if not isinstance(matrix, (Matrix, MatrixSymbol, ImmutableMatrix)):
+    if not isinstance(matrix, (Matrix, MatrixSymbol, ImmutableMatrix, FunctionMatrix)):
         raise TypeError("Transition probabilities either should "
```

Comment:
Matrix and ImmutableMatrix are both subclasses of MatrixBase so it's better to test against that. That also covers spares matrices. MatrixSymbol, FunctionMatrix and many more are subclasses of MatExpr. A better test would be:
```
if not isinstance(MatrixBase, MatExpr):
```